Subject: mm: page_alloc: Use migrate_disable() in drain_local_pages_wq()
From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Date: Thu Jul  2 14:27:23 2020 +0200

From: Sebastian Andrzej Siewior <bigeasy@linutronix.de>

drain_local_pages_wq() disables preemption to avoid CPU migration during
CPU hotplug and can't use cpus_read_lock().

Using migrate_disable() works here, too. The scheduler won't take the
CPU offline until the task left the migrate-disable section.

Use migrate_disable() in drain_local_pages_wq().

Signed-off-by: Sebastian Andrzej Siewior <bigeasy@linutronix.de>
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>


---
 mm/page_alloc.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)
---
--- a/mm/page_alloc.c
+++ b/mm/page_alloc.c
@@ -3164,9 +3164,9 @@ static void drain_local_pages_wq(struct
 	 * cpu which is alright but we also have to make sure to not move to
 	 * a different one.
 	 */
-	preempt_disable();
+	migrate_disable();
 	drain_local_pages(drain->zone);
-	preempt_enable();
+	migrate_enable();
 }
 
 /*
